metric.go

```diff
@@ -210,26 +233,72 @@ func (m *constMetric) Write(out *dto.Metric) errors.WithContext {
 	return nil
 }
 
-func makeLabelPairs(desc MetricDesc, labelValues []string) []*dto.LabelPair {
+func parseJsonLabels(desc MetricDesc, labels string) []*dto.LabelPair {
+	var userLabels []*dto.LabelPair
+	var jsonLabels map[string]string
+
+	config := desc.GlobalConfig()
+	maxJsonLabels := 0
```

Comment:
If `config == nil` (which will never happen AFAICT, so you probably don't need the check below) you'll actually end up with zero labels.